source/MissionAction.cpp

```diff
@@ -57,6 +57,60 @@ namespace {
 
 
 
+MissionAction::MissionDialog::MissionDialog(const ExclusiveItem<Phrase> &phrase):
+	dialogPhrase(phrase)
+{
+}
+
+
+
+MissionAction::MissionDialog::MissionDialog(const string &text):
+	dialogText(text)
+{
+}
+
+
+
+MissionAction::MissionDialog::MissionDialog(const DataNode &node)
+{
+	// Handle anonymous phrases
+	//    phrase
+	//       ...
+	if(node.Size() == 1 && node.Token(0) == "phrase")
+	{
+		dialogPhrase = ExclusiveItem<Phrase>(Phrase(node));
+		// Anonymous phrases do not support "to display"
+		return;
+	}
+
+	// Handle named phrases
+	//    phrase "A Phrase Name"
+	if(node.Size() == 2 && node.Token(0) == "phrase")
+		dialogPhrase = ExclusiveItem<Phrase>(GameData::Phrases().Get(node.Token(1)));
+
+	// Handle regular dialog text
+	//    "Some thrilling dialog that truly moves the player."
+	else
+	{
+		if(node.Size() > 1)
+			node.PrintTrace("Ignoring extra tokens.");
+		dialogText = node.Token(0);
+
+		// Prevent a corner case that breaks assumptions. Dialog text cannot be empty (that indicates a phrase).
+		if(dialogText.empty())
+			dialogText = '\t';
+	}
+
+	// Search for "to display" lines.
+	for(auto &child : node)
+		if(child.Size() != 2 || child.Token(0) != "to" || child.Token(1) != "display" || !child.HasChildren())
+			node.PrintTrace("Ignoring unrecognized dialog token");
+		else
+			condition.Load(child);
```

Comment:
```suggestion
	for(auto &child : node)
	{
		if(child.Size() != 2 || child.Token(0) != "to" || child.Token(1) != "display" || !child.HasChildren())
			node.PrintTrace("Ignoring unrecognized dialog token");
		else
			condition.Load(child);
	}
```
I know they are not necessary but please

Reply:
I don't have enough fingers and toes to count the number of times people have asked for the opposite change.